oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/Optimizer.kt

```diff
@@ -0,0 +1,18 @@
+package com.empowerops.volition.ref_oasis
+
+import kotlin.random.Random
+
+interface Optimizer {
+    fun generateInputs(inputs: List<Input>): Map<String, Double>
+}
+
+class RandomNumberOptimizer : Optimizer {
+    override fun generateInputs(inputs: List<Input>): Map<String, Double> = inputs.associate {
+        if(it.lowerBound == it.upperBound){
+            it.name to Random.nextDouble(it.lowerBound, it.upperBound+10.0)
```

Comment:
This seems pretty magical to me. Why not simply a `requires` statement here?

Reply:
Going to remove this for now